src/test/scala/org/broadinstitute/dsde/agora/server/webservice/ApiServiceSpec.scala

```diff
@@ -193,11 +201,30 @@ with AgoraTestData with BeforeAndAfterAll {
     }
   }
 
-  def handleError[T](deserialized: Deserialized[T], assertions: (T) => Unit) = {
-    if (status.isSuccess) {
-      if (deserialized.isRight) assertions(deserialized.right.get) else failTest(deserialized.left.get.toString)
-    } else {
-      failTest(response.message.toString)
+  "Agora" should "return only included fields in the entity" in {
+    Get(ApiUtil.Methods.withLeadingSlash + "?namespace=" + namespace1.get + "&name=" + name2.get + "&includedField=name&includedField=snapshotId") ~>
+      methodsService.queryRoute ~> check {
+      handleError(
+        entity.as[Seq[AgoraEntity]],
+        (entities: Seq[AgoraEntity]) =>
+          assert(entities === includeProjection(Seq(testEntity3WithId, testEntity4WithId, testEntity5WithId, testEntity6WithId, testEntity7WithId)))
+      )
+      assert(status === OK)
+    }
+  }
+
+  "Agora" should "not return excluded fields in the entity" in {
+    Get(ApiUtil.Methods.withLeadingSlash
+      + "?namespace=" + namespace1.get
+      + "&name=" + name2.get
+      + "&excludedField=synopsis&excludedField=documentation&excludedField=createDate&excludedField=payload") ~>
+      methodsService.queryRoute ~> check {
+      handleError(
+        entity.as[Seq[AgoraEntity]],
+        (entities: Seq[AgoraEntity]) =>
+          assert(entities === excludeProjection(Seq(testEntity3WithId, testEntity4WithId, testEntity5WithId, testEntity6WithId, testEntity7WithId)))
+      )
+      assert(status === OK)
     }
```

Comment:
We should be catching this ... but I can add a test that verifies the case class validation fails.